scripts/windows/setup/IotEdgeSecurityDaemon.ps1

```diff
@@ -641,52 +658,58 @@ function Remove-SecurityDaemonResources {
     else {
         Write-Verbose "$cmdErr"
     }
+    
+    $existingMobyRoots = $MobyDataRootDirectory, $MobyStaticDataRootDirectory | ? {test-path $_}
 
     if ($DeleteMobyDataRoot) {
-        if (Test-Path $MobyDataRootDirectory) {
+        foreach ($root in $existingMobyRoots) {
             try {
-                Write-Host "Deleting Moby data root directory '$MobyDataRootDirectory'..."
+                Write-Host "Deleting Moby data root directory '$root'..."
 
                 # Removing `$MobyDataRootDirectory` is tricky. Windows base images contain files owned by TrustedInstaller, etc
                 # Deleting them is a three-step process:
                 #
                 # 1. Take ownership of all files
-                Invoke-Native "takeown /r /skipsl /f ""$MobyDataRootDirectory"""
+                Invoke-Native "takeown /r /skipsl /f ""$root"""
 
                 # 2. Reset their ACLs so that they inherit from their container
-                Invoke-Native "icacls ""$MobyDataRootDirectory"" /reset /t /l /q /c"
+                Invoke-Native "icacls ""$root"" /reset /t /l /q /c"
 
                 # 3. Use cmd's `rd` rather than `Remove-Item` since the latter gets tripped up by reparse points, etc.
                 #    Prepend the path with `\\?\` since the layer directories have long names, so the paths usually exceed 260 characters,
                 #    and IoT Core's filesystem doesn't seem to automatically use (or even have) short names
-                Invoke-Native "rd /s /q ""\\?\$MobyDataRootDirectory"""
+                Invoke-Native "rd /s /q ""\\?\$root"""
 
-                Write-Verbose "Deleted Moby data root directory '$MobyDataRootDirectory'"
+                Write-Verbose "Deleted Moby data root directory '$root'"
             }
             catch {
                 Write-Verbose "$_"
-                Write-HostRed ("Could not delete Moby data root directory '$MobyDataRootDirectory'. Please reboot " +
+                Write-HostRed ("Could not delete Moby data root directory '$root'. Please reboot " +
                     'your device and run `Uninstall-SecurityDaemon` again with `-Force`.')
                 $success = $false
             }
         }
     }
     else {
-        Write-Host "Not deleting Moby data root directory '$MobyDataRootDirectory' since -DeleteMobyDataRoot was not specified."
+        Write-Host "Not deleting Moby data root directory since -DeleteMobyDataRoot was not specified."
     }
+    
+    $existingMobyInstallations = $MobyInstallDirectory, $MobyStaticInstallDirectory | ? {test-path $_}
```

Comment:
`?{ Test-Path $_ }`

Although this isn't really necessary since `PathNotFound` gets handled below anyway, so you could just take it out.

Reply:
The goal is to not express the directory missing as an error. I guess the other potential issue is if these two are the same... we would always report failure on the second iteration.

Reply:
Makes sense. Leave it in and just update the formatting then.

Reply:
I refactored this a bit but I'm not sure if it's meaningful since we silently continue on failure.